__mocks__/has-yarn.js

```diff
@@ -0,0 +1,4 @@
+let hasYarn = false;
+
+module.exports = () => hasYarn;
+module.exports.__setHasYarn = value => (hasYarn = !!value);
```

Comment:
Why all those ```__``` and ```!!``` ?

Reply:
The `__` is a convention that's [documented here](https://facebook.github.io/jest/docs/en/manual-mocks.html). The `!!` is just to coerce the value to boolean.